chalice/cli/reload.py

```diff
@@ -0,0 +1,71 @@
+import os
+import subprocess
+import sys
+import threading
+import time
+import types
+
+import six
+from typing import Optional, Dict, Type  # noqa
+
+from chalice.constants import AUTORELOAD_INTERVAL
+
+
+class Reloader(threading.Thread):
+    def __init__(self, autoreload=True):
+        # type: (bool) -> None
+        super(Reloader, self).__init__()
+        self.autoreload = autoreload
+        self.triggered = False
```

Comment:
I recognize that this is whether or not the process has ever reloaded. But can you give some clarification on why this is important to keep track of?

Reply:
We need to have additional behavior when interrupting for the reloaded process in windows (raise SystemExit).